react-hook/cra-implementation/src/index.js

```diff
@@ -0,0 +1,107 @@
+import React, { useEffect, useRef, useState } from "react";
+import ReactDOM from "react-dom";
+import "./index.css";
+
+const DecimalClock = () => {
+  const frameRequest = useRef(null);
+  const [date, setDate] = useState(new Date());
+  const [fractionOfDay, setFractionOfDay] = useState(null);
+  const [timeString, setTimeString] = useState(null);
+
+  const update = () => {
+    setDate(new Date());
+    frameRequest.current = requestAnimationFrame(update);
+  };
+
+  useEffect(() => {
+    frameRequest.current = requestAnimationFrame(update);
+
+    return () => {
+      cancelAnimationFrame(frameRequest.current);
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    const timeOfDay =
+      (date.getHours() * 3600 + date.getMinutes() * 60 + date.getSeconds()) *
+        1000 +
+      date.getMilliseconds();
+    setFractionOfDay(timeOfDay / 86400_000);
+  }, [date]);
+
+  useEffect(() => {
+    setTimeString(
+      `${Math.floor(fractionOfDay * 10)}:${Math.floor(
+        (fractionOfDay * 1_000) % 100
+      )
+        .toString()
+        .padStart(2, "0")}:${Math.floor((fractionOfDay * 100_000) % 100)
+        .toString()
+        .padStart(2, "0")}`
+    );
+  }, [fractionOfDay]);
+
+  return (
+    <>
+      <h2>Decimal Clock</h2>
+      <div id="clock-face">
+        {[...Array(10)].map((_, i) => (
```

Comment:
`[...Array(10)]` can be declared as const outside component body to prevent it from changing reference on every render